mitmproxy/proxy/layers/quic.py

```diff
@@ -380,6 +381,12 @@ def initialize_replacement(peer_cid: bytes) -> None:
             raise ValueError("Invalid ClientHello data.") from e
     except QuicConnectionError as e:
         raise ValueError(e.reason_phrase) from e
+
+    # Fragmented Clienthellos seem to produce valid replies whereas invalid Clienthellos don't
+    # We delay sending these replies until we receive the entire client hello
+    if quic.datagrams_to_send(now=0.1):
+        return None
```

Comment:
One case that we miss here is that if the second fragment happens to be invalid i.e. we receive a valid first fragment and a corrupt second fragment (and the entire Clienthello is sent across two fragments). Since the entire clienthello is invalid, but the first fragment is valid, we don't get valid Clienthello but we receive a non empty value from `datagrams_to_send` and we end up waiting for more fragments instead of raising an error.

One solution to this problem is that we could make `recv_buffer` a `list[bytes]` and add new fragments as a whole to the start of the buffer (and then convert to `bytes` while passing to `quic_parse_client_hello`). Doing so, we raise an error as the first packet being read from the buffer is invalid (for the above case) (If it were valid we will return a Clienthello as we can also parse out-of-order fragments). But would this added complexity be a good tradeoff for a seemingly rare edgecase?